cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -454,29 +313,181 @@ angular.module(PKG.name + '.commons')
       };
     }
 
-    function formatConnections() {
-      var connections = [];
-      angular.forEach(vm.instance.getConnections(), function (conn) {
-        connections.push({
-          from: conn.sourceId,
-          to: conn.targetId
-        });
+    function addConnection(newConnObj) {
+      $scope.connections.push({
+        from: newConnObj.sourceId,
+        to: newConnObj.targetId
+      });
+      DAGPlusPlusNodesActionsFactory.setConnections($scope.connections);
+    }
+
+    function removeConnection(detachedConnObj, updateStore = true) {
+      var detachedConnection = _.find($scope.connections, function (conn) {
+        return conn.from === detachedConnObj.sourceId && conn.to === detachedConnObj.targetId;
+      });
+      if (detachedConnection) {
+        $scope.connections.splice($scope.connections.indexOf(detachedConnection), 1);
+      }
+      if (updateStore) {
+        DAGPlusPlusNodesActionsFactory.setConnections($scope.connections);
+      }
+    }
+
+    function moveConnection(moveInfo) {
+      let oldConnection = {
+        sourceId: moveInfo.originalSourceId,
+        targetId: moveInfo.originalTargetId
+      };
+      // don't need to call addConnection for the new connection, since that will be done
+      // automatically as part of the 'connection' event
+      removeConnection(oldConnection, false);
+    }
+
+    vm.removeSelectedConnections = function() {
+      if (selectedConnections.length === 0 || vm.isDisabled) { return; }
+
+      angular.forEach(selectedConnections, function (selectedConnectionObj) {
+        vm.instance.unbind('connectionDetached');
+        removeContextMenuEventListener(selectedConnectionObj);
+        vm.instance.detach(selectedConnectionObj);
+        removeConnection(selectedConnectionObj, false);
+        vm.instance.bind('connectionDetached', removeConnection);
+      });
+      selectedConnections = [];
+      DAGPlusPlusNodesActionsFactory.setConnections($scope.connections);
+    };
+
+    function selectEndpointOrConnection(seletectedObj) {
```

Comment:
nit: This function is always selecting/de-selecting connections. The endpoint behavior is still selecting/de-selecting connections (a side effect of clicking on endpoints). Probably easier to refer this as `selectConnections`? Just a thought.